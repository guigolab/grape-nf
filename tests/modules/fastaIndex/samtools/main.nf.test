nextflow_process {

    name "Test Process fastaIndex"
    script "modules/fastaIndex/samtools/main.nf"
    process "fastaIndex"

    tag "module"

    test("Should produce the fasta index (compressed genome)") {

        when {
            process {
                """
                input[0] = "${baseDir}/data/genome.fa.gz"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should produce the fasta index (uncompressed genome)") {

        when {
            process {
                """
                input[0] = "${baseDir}/data/genome.fa"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
