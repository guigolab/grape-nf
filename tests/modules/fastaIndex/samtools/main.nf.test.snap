{
    "Should produce the fasta index (compressed genome)": {
        "content": [
            {
                "0": [
                    "genome.fa.fai:md5,12d36d122df8dfe8fcfc96d78f785328"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T18:59:06.37067"
    },
    "Should produce the fasta index (uncompressed genome)": {
        "content": [
            {
                "0": [
                    "genome.fa.fai:md5,12d36d122df8dfe8fcfc96d78f785328"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T18:59:15.457162"
    }
}